src/pages/Signup/index.js

```diff
@@ -1,84 +1,70 @@
 import RegistrationForm from '../../components/RegistrationForm'
-import React from 'react'
+import { useState } from 'react'
+import { useHistory } from "react-router-dom"
 import axios from 'axios'
+function Signup(){
+    const [userSignUp, setUserSignUp] = useState({ email : '', password : ''})
+    const [confirmedPassword, setConfirmedPassword] = useState('')
 
+    const [showPasswordWarning, setShowPasswordWarning] = useState(false)
+    const [showUserWarning, setShowUserWarning] = useState(false)
 
-class Signup extends React.Component{
-  state={
-    email: '',
-    password: '',
-    confirmedPassword:'',
-    showPasswordWarning:false,
-    showUserWarning:false,
-  }
+    const SIGNUP_ENDPOINT=process.env.REACT_APP_SERVER_URL+'/users/signup'
 
-  handleSubmit = async e =>{
-    e.preventDefault()
-    const { password, confirmedPassword } = this.state
+    let history = useHistory()
 
-    this.setState({
-      showPasswordWarning:false,
-      showUserWarning:false,
-    })
+    function handleSubmit (e){
+        e.preventDefault()
+        setShowPasswordWarning(false)
+        setShowUserWarning(false)
 
-    if( password === confirmedPassword ){
-      try {
-        const { data } = await axios({
-          method: 'POST',
-          baseURL: process.env.REACT_APP_SERVER_URL,
-          url: '/users/signup',
-          data: this.state
-        })
+        if( userSignUp.password === confirmedPassword ){
 
-        localStorage.setItem('user', data.email)
-        this.props.history.push('/welcome')
-        
-      } catch(error) {
-          this.setState({
-          showUserWarning : true,
-          email:'',
-          password: '',
-          confirmedPassword:''
-          })
+            axios.post(SIGNUP_ENDPOINT, userSignUp)
+            .then( (res) => {
+                const { email } = res.data
+                localStorage.setItem('user', email)
+                history.push('/welcome')
+            })
+            .catch(function (error) {
+                console.log(error)
+                setUserSignUp({ email : '', password : ''})
+                setConfirmedPassword('')
+                setShowUserWarning(true)
+
+            })
+        }else{
+            setShowPasswordWarning(true)
+            setUserSignUp({ email : '', password : ''})
+            setConfirmedPassword('')
         }
-    } else{
-      this.setState({
-          showPasswordWarning : true,
-          password: '',
-          confirmedPassword:''
-        })
-    } 
-  }
+    }
+
 
-  handleChange = e =>{
-    const { name, value } = e.target
-    this.setState({
-      [name] : value,
-    })
-  }
+    function handleChange (e){
+        const { name, value } = e.target
+        setUserSignUp({...userSignUp, [name] : value})
+    }
 
+    function handleConfirmPassword (e){
+        const { value } = e.target
+        setConfirmedPassword(value)
```

Comment:
Alguna razón para manejar la confirmación de la contraseña por aparte?

Reply:
por qué se modifican ambos campos a la vez? esto no debería ocurrir a menos que: los dos inputs tengan el mismo name, o ambos esten mappeando el mismo value del estado.
El problema acá es que en una parte estás combinando 2 estados pero otro lo manejas independiente. Como dicen por ahí, o todos en la cama o todos en el suelo

Reply:
@simonhoyos el problema que te comento sucede porque lo he interpretado de la siguiente manera:
```javascript
  function handleChange (e){
        const { name, value } = e.target
        setUserSignUp({...userSignUp, [name] : value})
      setConfirmedPassword(value)
    }
```
¿Qué sugieres para resolver ese issue? Me siento un poco confundido, dado que no recomiendas trabajar el `handle` del _confirmedPassword_ por aparte, sino todos en el mismo `handle`.

Reply:
claro, el cambio simultaneo ocurre porque estás ejecutando las dos funciones. 

```javascript
const [ userSignUp, setUserSignUp ] = useState({
  email: '',
  password: '',
  confirmedPassword: '',
})

function handleChange(e) {
  const { name, value } = e.target
  setUserSignUp({ ...userSignUp, [name]: value })
}
```

Reply:
Please, check the last set of changes in last commit. Thanks in advance.